src/components/visualizations/chart/colorFactory.ts

```diff
@@ -260,7 +262,8 @@ function getAtributeColorAssignment(
     return uniqItems.map((headerItem) => {
         const mappedColor = getColorFromMapping(headerItem, colorMapping);
 
-        const color: IColorItem = mappedColor ? mappedColor
+        const color: IColorItem = mappedColor && isValidMappedColor(mappedColor, colorPalette)
```

Comment:
just thinking, if we would have 
`if(!mappedColor) { return false; }` at the very beginining of `isValidMappedColor` method, we could avoid having `mappedColor &&` here and also on other places (the same here https://github.com/gooddata/gooddata-react-components/pull/674/files#diff-934a24e28244a84cb0a91e27b0b8aee2R442)